samples/jit_tutorial/main.cpp

```diff
@@ -0,0 +1,238 @@
+/*******************************************************************************
+ *
+ *   JIT tutorial sample: branching and graph reuse.
+ *
+ *   Demonstrates:
+ *   - Tape: re-records and therefore follows normal C++ control flow per run.
+ *   - JIT: graph is recorded once; plain C++ `if` is baked in at record time.
+ *   - ABool::If: records a conditional node so the branch can vary at runtime.
+ *
+ *   This file is part of XAD, a comprehensive C++ library for
+ *   automatic differentiation.
+ *
+ *   Copyright (C) 2010-2025 Xcelerit Computing Ltd.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU Affero General Public License as published
+ *   by the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU Affero General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Affero General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+
+#include <XAD/XAD.hpp>
+
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+namespace
+{
+
+template <class AD>
+AD piecewise_plain_if(const AD& x)
+{
+    // Normal C++ control flow: decision is made immediately, based on the current value.
+    if (xad::value(x) < 2.0)
+        return 1.0 * x;
+    return 7.0 * x;
+}
+
+template <class AD>
+AD piecewise_abool_if(const AD& x)
+{
+    // Trackable control flow for JIT: record both branches and select at runtime.
+    auto cond = xad::less(x, 2.0);
```

Comment:
Maybe add a comment here that `xad::less` returns an `ABool`, or make it explicit?